node.go

```diff
@@ -602,3 +602,159 @@ func sortFields(fields []Field) {
 func compareFields(a, b Field) int {
 	return strings.Compare(a.Name(), b.Name())
 }
+
+func RewriteSchema(schema *Schema, replace func(path []string, node Node) Node) *Schema {
```

Comment:
If we want to keep this attached to the concrete `Schema` type I would recommend making it a method:
```go
func (s *Schema) Rewrite(replace func(path []string, node Node) Node) *Schema {
  ...
}
```
Alternatively, we could decouple this from `Schema` and have it work on `Node`:
```go
func Rewrite(root Node, replace func(path []string, node Node) Node) Node {
  ...
}
```